diff -Naru -x .git ../Disruptor-cpp/Disruptor/CMakeLists.txt ./Disruptor/CMakeLists.txt
--- ../Disruptor-cpp/Disruptor/CMakeLists.txt	2021-02-24 14:08:04.000000000 -0500
+++ ./Disruptor/CMakeLists.txt	2021-02-24 14:23:55.000000000 -0500
@@ -22,6 +22,7 @@
     Sequence.cpp
     SequenceGroups.cpp
     SleepingWaitStrategy.cpp
+    SpinCountBackoffWaitStrategy.cpp
     SpinWait.cpp
     SpinWaitWaitStrategy.cpp
     stdafx.cpp
@@ -100,6 +101,7 @@
     Sequencer.h
     SingleProducerSequencer.h
     SleepingWaitStrategy.h
+    SpinCountBackoffWaitStrategy.h
     SpinWait.h
     SpinWaitWaitStrategy.h
     ThreadPerTaskScheduler.h
diff -Naru -x .git ../Disruptor-cpp/Disruptor/SpinCountBackoffWaitStrategy.cpp ./Disruptor/SpinCountBackoffWaitStrategy.cpp
--- ../Disruptor-cpp/Disruptor/SpinCountBackoffWaitStrategy.cpp	1969-12-31 19:00:00.000000000 -0500
+++ ./Disruptor/SpinCountBackoffWaitStrategy.cpp	2021-02-24 14:23:44.000000000 -0500
@@ -0,0 +1,71 @@
+#include "stdafx.h"
+#include "SpinCountBackoffWaitStrategy.h"
+#include "BlockingWaitStrategy.h"
+
+#include <ostream>
+#include <iostream>
+
+#include "ISequenceBarrier.h"
+#include "Sequence.h"
+
+
+namespace Disruptor
+{
+
+    SpinCountBackoffWaitStrategy::SpinCountBackoffWaitStrategy()
+    {
+        this->SPIN_TRIES = 10000;
+        this->fallbackStrategy = std::make_shared< BlockingWaitStrategy >();
+    }
+
+    SpinCountBackoffWaitStrategy::SpinCountBackoffWaitStrategy(std::uint32_t spinTries,
+                                                               std::shared_ptr<IWaitStrategy> fallbackStrategy)
+    {
+        this->SPIN_TRIES = spinTries;
+        if (spinTries < 1) {
+            this->SPIN_TRIES = 1;
+        }
+
+        this->fallbackStrategy = fallbackStrategy;
+        if (fallbackStrategy == nullptr) {
+            this->fallbackStrategy = std::make_shared< BlockingWaitStrategy >();
+        }
+    }
+
+
+    std::int64_t SpinCountBackoffWaitStrategy::waitFor(std::int64_t sequence,
+                                                       Sequence& cursor,
+                                                       ISequence& dependentSequence,
+                                                       ISequenceBarrier& barrier)
+    {
+        std::int64_t availableSequence;
+        std::uint32_t counter = SPIN_TRIES;
+
+        do {
+            if ((availableSequence = dependentSequence.value()) >= sequence) {
+                return availableSequence;
+            }
+
+            if (0 == --counter) {
+                return fallbackStrategy->waitFor(sequence, cursor, dependentSequence, barrier);
+            }
+            //std::cout << " " <<  counter;
+            barrier.checkAlert();
+            //spinWait.spinOnce();
+
+        } while (true);
+    }
+
+    void SpinCountBackoffWaitStrategy::signalAllWhenBlocking()
+    {
+        fallbackStrategy->signalAllWhenBlocking();
+    }
+
+    void SpinCountBackoffWaitStrategy::writeDescriptionTo(std::ostream& stream) const
+    {
+        stream << "SpinCountBackoffWaitStrategy";
+    }
+
+
+} // namespace Disruptor
+
diff -Naru -x .git ../Disruptor-cpp/Disruptor/SpinCountBackoffWaitStrategy.h ./Disruptor/SpinCountBackoffWaitStrategy.h
--- ../Disruptor-cpp/Disruptor/SpinCountBackoffWaitStrategy.h	1969-12-31 19:00:00.000000000 -0500
+++ ./Disruptor/SpinCountBackoffWaitStrategy.h	2021-02-24 14:23:44.000000000 -0500
@@ -0,0 +1,57 @@
+#pragma once
+
+#include <boost/thread.hpp>
+
+#include "Disruptor/IWaitStrategy.h"
+
+
+namespace Disruptor
+{
+
+    /**
+     * <p>The SpinCountBackoffWait strategy spins for a given number of times then
+     * waits using the configured fallback WaitStrategy.</p>
+     * This combination wait strategy worked well in Java with the fallback being BlockingWaitStrategy,
+     * so that's the default here.
+     *
+     * @author timmer
+     * @date 2/18/2020
+     */
+    class SpinCountBackoffWaitStrategy : public IWaitStrategy
+    {
+
+    public:
+
+        /** Constructor defaulting to 10,000 spins and BlockingWaitStrategy when finished spinning. */
+        SpinCountBackoffWaitStrategy();
+
+        /**
+         * Constructor specifying number of spins and the back up wait strategy it switches to after spinning.
+         * @param spinTries number of spins.
+         * @param fallbackStrategy back up wait strategy.
+         */
+        SpinCountBackoffWaitStrategy(std::uint32_t spinTries, std::shared_ptr<IWaitStrategy> fallbackStrategy);
+
+        /**
+         * \see IWaitStrategy::waitFor
+         */
+        std::int64_t waitFor(std::int64_t sequence,
+                             Sequence& cursor,
+                             ISequence& dependentSequence,
+                             ISequenceBarrier& barrier) override;
+
+        /**
+         * \see IWaitStrategy::signalAllWhenBlocking
+         */
+        void signalAllWhenBlocking() override;
+
+        void writeDescriptionTo(std::ostream& stream) const override;
+
+    private:
+
+        std::uint32_t SPIN_TRIES;
+        std::shared_ptr<IWaitStrategy> fallbackStrategy;
+
+    };
+
+} // namespace Disruptor
